### PR TITLE
Update to latest copy of ResolveNuGetPackageAssets.

### PR DESCRIPTION
**Pull request by ericstj**
This version handles transitive dependencies via the .lock.json.  We aren't yet using the .lock.json for asset selection. /cc @weshaggard @jasonmalinowski

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks/ResolveNuGetPackageAssets.cs [hunk 1]
```diff
@@ -302,10 +291,10 @@ private IList<ITaskItem> GetTaskItemsFromLibrary(Library library, ContentPattern
...
             if (group == null)
             {
-                return taskItems;
+                return null;
             }
 
-            foreach (var item in group.Items)
+            foreach (var item in group.Items.Where(i => Path.GetFileName(i.Path) != "_._"))
             {
                 var taskItem = new TaskItem(Path.Combine(Path.GetDirectoryName(library.Path), item.Path.Replace('/', '\\')));
```
Comment (akoeplinger) [inline src/Microsoft.DotNet.Build.Tasks/ResolveNuGetPackageAssets.cs]: unrelated to this PR, but can you replace `\\` with `Path.DirectorySeparatorChar`? it should help Unix portability.

**Comment by ericstj** [comment]
This will enable me to do this: https://github.com/ericstj/corefx/commit/bd7ee140a2a8470293a7e008929194d5fe918aa8

**Comment by ericstj** [comment]
Discussed with @weshaggard and have verbal sign-off.

File: src/Microsoft.DotNet.Build.Tasks/PropertyDefinitions.cs [hunk 2]
```diff
@@ -86,12 +86,12 @@ public PropertyDefinitions()
 
         private readonly ContentPropertyDefinition _assembly = new ContentPropertyDefinition
         {
-            FileExtensions = { ".dll" }
+            FileExtensions = { ".dll", "_._" }
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks/PropertyDefinitions.cs]: What is "_._"?

Comment (ericstj) [inline src/Microsoft.DotNet.Build.Tasks/PropertyDefinitions.cs]: _._ is a placeholder file.  It's a convention used in nuget to indicate that the particular pivot is supported but no assets apply.

**Comment by weshaggard** [comment]
I've not seen this in action yet but the changes look reasonable to me.

**Pull request merged**